safenode/src/network/api.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use std::collections::HashSet;
+
+use super::{Result, SafeRequest, SafeResponse};
+use futures::{
+    channel::{mpsc, oneshot},
+    SinkExt,
+};
+use libp2p::{request_response::ResponseChannel, Multiaddr, PeerId};
+use xor_name::XorName;
+
+use super::command::CmdToSwarm;
+
+#[derive(Clone)]
+/// API to interact with the underlying Swarm
+pub struct NetworkApi {
+    pub(super) sender: mpsc::Sender<CmdToSwarm>,
+}
+
+impl NetworkApi {
+    //  Listen for incoming connections on the given address.
+    pub async fn start_listening(&mut self, addr: Multiaddr) -> Result<()> {
+        let (sender, receiver) = oneshot::channel();
```

Comment:
sender and `self.sender` are confusing here. Can we name one of them a bit differently for clarity?